scripts/download_osm_data.py

```diff
@@ -425,338 +70,44 @@ def convert_iso_to_geofk(iso_code, iso_coding=True, convert_dict=iso_to_geofk_di
-        ["NG","ZA"], downloading osm data for Nigeria and South Africa
-        ["SNGM"], downloading data for Senegal&Gambia shape
-        ["NG","ZA","NG"], won't distort result.
-
-    Returns
-    -------
-    full_codes_list : list
-        Example ["NG","ZA"]
-    """
-
-    full_codes_list = [convert_iso_to_geofk(c_code) for c_code in set(country_list)]
-
-    return full_codes_list
-
-
 if __name__ == "__main__":
     if "snakemake" not in globals():
-        from _helpers import mock_snakemake
+        from _helpers import mock_snakemake, sets_path_to_root
 
         os.chdir(os.path.dirname(os.path.abspath(__file__)))
         snakemake = mock_snakemake("download_osm_data")
+        sets_path_to_root("pypsa-earth")
     configure_logging(snakemake)
 
-    # Required to set path to pypsa-earth
-    sets_path_to_root("pypsa-earth")
-
-    # ["substation", "generator", "line", "cable", "tower"]
-    feature_list = ["substation", "generator", "line", "cable"]
-
-    # get list of countries into geofabrik convention; expected iso norm in input
+    store_path_data = Path.joinpath(Path().cwd(), "data", "osm")
+    store_path_resources = Path.joinpath(Path().cwd(), "resources", "osm", "raw")
     country_list = country_list_to_geofk(snakemake.config["countries"])
-    output_files = snakemake.output  # output snakemake
-    nprocesses = snakemake.config.get(
-        "download_osm_data_nprocesses", 1
-    )  # number of threads
-
-    # get the default and metric crs of the workflow
-    geo_crs = snakemake.config["crs"]["geo_crs"]
-    distance_crs = snakemake.config["crs"]["distance_crs"]
 
-    # Set update # Verify = True checks local md5s and pre-filters data again
-    process_data(
-        feature_list,
-        country_list,
-        output_files,
-        geo_crs,
-        distance_crs,
-        iso_coding=True,
+    eo.get_osm_data(
+        primary_name="power",
+        region_list=country_list,
+        feature_list=["substation", "line", "cable", "generator"],
         update=False,
-        verify=False,
-        nprocesses=nprocesses,
+        mp=True,
+        data_dir=store_path_data,
+        out_format=["csv", "geojson"],
+        out_aggregate=True,
     )
+
+    out_path = Path.joinpath(store_path_data, "out")
+    names = ["generators", "cables", "lines", "substations"]
+    format = ["csv", "geojson"]
+
+    # Create file if not existing
+    for name in names:
+        for f in format:
+            filename = Path.joinpath(out_path, f"all_{name}.{f}")
+            if not Path.exists(filename):
+                _logger.info(f"{filename} does not exist, create empty file")
+                open(filename, "w").close()
+
+            # Move and rename
+            old_path = Path.joinpath(out_path, f"all_{name}.{f}")
+            new_path = Path.joinpath(store_path_resources, f"africa_all_{name}.{f}")
+            _logger.info(f"Create {old_path} and move to {new_path}")
+            shutil.move(old_path, new_path)
```

Comment:
We could remove this section in case we:
- Read directly from the `pypsa-earth/data/osm/out` folder
- Rename the files in the Snakefile e.g. from `africa_all_generators.csv` to `all_generators.csv`

Reply:
Any opinion here? @davide-f 
I think the rename is due to anyway (there should be no 'africa' in the name anymore), but I am not sure about reading the files form `data/osm/out` instead from `resources/osm/`

Reply:
@pz-max I am ok with removing the africa_ because we want to be general.
Regarding whether to move the raw outputs into data, I think we need to adopt a policy according to the others as well.

At the moment, I'd keep things as is as otherwise there may conflicts everywhere in the PRs and this is a detail I'd say.

Regarding earth-osm, I have few questions:
- can multiple executions of earth-osm work in parallel? before because of esy-* that was impossible: all parallel threads would have worked on a common temp file; what now?
- Linked to above, the download_osm_data may be done in a snakemake-like way, where the type of resource may be a wildcard. That could be really useful for the -sec and distribution packages if done well

Can be left to further PRs though

Reply:
@mnm-matin feel free to correct me. The current implementation includes an option `mp=bool` which stands for multiprocessing. It works exactly like esy-osm. It will use all available computational resources. (e.g. set by snakemake)
- When you speak about resources, you mean the type of infrastructure? Sure, once we have a nice interface to also extract other infrastructure, we might only need to define a config and can automate stuff

Reply:
So, having parallel computing within the tool is cool.
My concern is what if two parallel processes (e.g. the same rule with different wildcard) operate using the same data_dir?
Are there differences if they work on the same or different countries?

Depending these points, it may or may not be possible to use wildcards easily with earth-osm

Reply:
Not sure about this. We can also deactivate parallel processes within earth-osm (it's an option `mp=False`)